### PR TITLE
(FM-8500) Create type for vrf

### PR DESCRIPTION
**Pull request by MaxMagill**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1294,6 +1295,38 @@ namevar
 
 The name of the tacacs server group.
 
+##### vrf
+
+Configure a vrf.
```
Comment (DavidS) [inline README.md]: please spell `VRF` consistently as a acronym throughout the docs.

File: lib/puppet/type/vrf.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        type:   'String',
+        desc:   'Name of vrf instance',
+        behaviour: :namevar
+      },
+      route_distinguisher: {
+        type:   'Optional[String]',
+        desc:   'Address qualifier for the vrf, used to amintain uniqueness among identical routes'
+      },
+      route_targets: {
+        type:   'Optional[Array[Tuple[enum["export", "import", "both"], String]]]',
```
Comment (DavidS) [inline lib/puppet/type/vrf.rb]: `enum` needs to be title-cased (`Enum`)

File: lib/puppet/type/vrf.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
+require_relative '../../puppet_x/puppetlabs/netdev_stdlib/check'
+if PuppetX::NetdevStdlib::Check.use_old_netdev_type
+  Puppet::Type.newtype(:syslog_settings) do
```
Comment (DavidS) [inline lib/puppet/type/vrf.rb]: There is no need to support legacy modules for this type. Please remove everything except the rsapi type here.

Create a temporary new module and use `pdk new provider` there to get a skeleton type and unit tests for this.

File: lib/puppet/type/vrf.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+require_relative '../../puppet_x/puppetlabs/netdev_stdlib/check'
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: the require_relative is uneeded

Comment (david22swan) [reply lib/puppet/type/vrf.rb]: Also, need's to be realigned.

File: README.md [hunk 5]
```diff
@@ -1294,6 +1295,38 @@ namevar
...
+
+Optional[String]
+
+Address qualifier for the VRF, used to amintain uniqueness among identical routes.
+
+##### `route_target`
+
+Optional[Array[Tuple[Enum["export", "import", "both"], String]]]
+
+Address qualifier for the VRF, used to share routes between VRF.
```
Comment (david22swan) [inline README.md]: This sentence is confusing, the second `VRF` should be `VRFs`

File: README.md [hunk 6]
```diff
@@ -54,6 +54,7 @@ types implemented in this module.
 * [`tacacs_global`](#tacacs_global): Configure global tacacs settings
 * [`tacacs_server`](#tacacs_server): Configure a tacacs server
 * [`tacacs_server_group`](#tacacs_server_group): Configure a tacacs server group
+* [`vrf`](#vrf): Configure a vrf
```
Comment (david22swan) [inline README.md]: The 3rd `vrf` should be `VRF`

File: lib/puppet/type/vrf.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        type:   'Optional[String]',
+        desc:   'Address qualifier for the vrf, used to amintain uniqueness among identical routes'
+      },
+      route_targets: {
+        type:   'Optional[Array[Tuple[Enum["export", "import", "both"], String]]]',
+        desc:   "Address qualifier for the vrf, used to share routes between vrfs"
+      },
+      import_map: {
+        type:   'Optional[String]',
+        desc:   'Associates a route map with the vrf'
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: should be `VRF`

File: lib/puppet/type/vrf.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        desc:   'Name of vrf instance',
+        behaviour: :namevar
+      },
+      route_distinguisher: {
+        type:   'Optional[String]',
+        desc:   'Address qualifier for the vrf, used to amintain uniqueness among identical routes'
+      },
+      route_targets: {
+        type:   'Optional[Array[Tuple[Enum["export", "import", "both"], String]]]',
+        desc:   "Address qualifier for the vrf, used to share routes between vrfs"
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: Should be `VRF` and `VRFs`

File: lib/puppet/type/vrf.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    features: ['canonicalize','simple_get_filter'] + ( Puppet::Util::NetworkDevice.current.nil? ? [] : ['remote_resource'] ),
+    attributes: {
+      name: {
+        type:   'String',
+        desc:   'Name of vrf instance',
+        behaviour: :namevar
+      },
+      route_distinguisher: {
+        type:   'Optional[String]',
+        desc:   'Address qualifier for the vrf, used to amintain uniqueness among identical routes'
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: Should be `VRF`

File: lib/puppet/type/vrf.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+
+require 'puppet/resource_api'
+  Puppet::ResourceApi.register_type(
+    name: 'vrf',
+    docs: 'Configure vrf settings',
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: Should be `VRF`

File: lib/puppet/type/vrf.rb [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    features: ['canonicalize','simple_get_filter'] + ( Puppet::Util::NetworkDevice.current.nil? ? [] : ['remote_resource'] ),
+    attributes: {
+      name: {
+        type:   'String',
+        desc:   'Name of vrf instance',
```
Comment (david22swan) [inline lib/puppet/type/vrf.rb]: Should be `the VRF`

**Pull request merged**